common/djangoapps/student/models/user.py

```diff
@@ -1426,29 +1442,46 @@ def _cert_name(self, course_name, cert_mode):
+        default_cert_name = self.MODE_TO_CERT_NAME.get(
+            cert_mode, _('{platform_name} Certificate for {course_name}')
+        )
         # Look for an override of the certificate name in the SOCIAL_SHARING_SETTINGS setting
-        share_settings = configuration_helpers.get_value('SOCIAL_SHARING_SETTINGS', settings.SOCIAL_SHARING_SETTINGS)
-        cert_name = share_settings.get('CERTIFICATE_LINKEDIN_MODE_TO_CERT_NAME', {}).get(cert_mode, default_cert_name)
+        cert_name = self.share_settings.get(
+            'CERTIFICATE_LINKEDIN_MODE_TO_CERT_NAME', {}
+        ).get(cert_mode, default_cert_name)
 
         return cert_name.format(
             platform_name=configuration_helpers.get_value('platform_name', settings.PLATFORM_NAME),
             course_name=course_name
         )
 
-    def _organization_information(self):
+    def _organization_information(self, course=False):
         """
-        Returns organization information for use in the URL parameters for add to profile.
+        Returns organization information for use in the URL parameters for add to
+        profile. By default when sharing to LinkedIn, Platform Name and/or Platform
+        LINKEDIN_COMPANY_ID will be used. If Course specific Organization Name is
+        prefered when sharing Certificate to linkedIn the flag for that
+        CERTIFICATE_LINKEDIN_DEFAULTS_TO_COURSE_ORGANIZATION_NAME should be set
+        to True alongside other LinkedIn settings
 
         Returns:
-            dict: Either the organization ID on LinkedIn or the organization's name
+            dict: Either the organization ID on LinkedIn, the organization's name or
+                organization name associated to a specific course
                 Will be used to prefill the organization on the add to profile action.
         """
-        org_id = configuration_helpers.get_value('LINKEDIN_COMPANY_ID', self.company_identifier)
-        # Prefer organization ID per documentation at https://addtoprofile.linkedin.com/
-        if org_id:
-            return {'organizationId': org_id}
-        return {'organizationName': configuration_helpers.get_value('platform_name', settings.PLATFORM_NAME)}
+        prefer_course_organization_name = self.share_settings.get(
+            'CERTIFICATE_LINKEDIN_DEFAULTS_TO_COURSE_ORGANIZATION_NAME', False
+        )
+        if (prefer_course_organization_name and (course is not False)):
+            return {"organizationName": course.display_organization}
+        else:
+            org_id = configuration_helpers.get_value(
+                "LINKEDIN_COMPANY_ID", self.company_identifier
+            )
+            # Prefer organization ID per documentation at https://addtoprofile.linkedin.com/
+            if org_id:
+                return {"organizationId": org_id}
+            return {'organizationName': configuration_helpers.get_value('platform_name', settings.PLATFORM_NAME)}
```

Comment:
Nit: I would write it like this, because you're already returning in the `if` branch, you don't need an `else`. And this way, at least personally, it's easier to view the code.
```suggestion

        org_id = configuration_helpers.get_value(
            "LINKEDIN_COMPANY_ID", self.company_identifier
        )
        # Prefer organization ID per documentation at https://addtoprofile.linkedin.com/
        if org_id:
            return {"organizationId": org_id}
        return {'organizationName': configuration_helpers.get_value('platform_name', settings.PLATFORM_NAME)}
```